### PR TITLE
Refactor the build system and deploy process

### PR DESCRIPTION
**Pull request by cal-smith**
- everything builds into `dist` directories
- all `.d.ts` files should now have accompanying `.js` files
- simplified the folder structures
- angular should build correctly
- added `.umd.js` where appropriate
- built `style.scss` to `style.css` for core
- removed `style.scss` from `index.ts` for core
- other, small changes

### CODE REVIEW

File: packages/vue/package.json [hunk 1]
```diff
@@ -4,13 +4,14 @@
   "description": "Carbon charting components for Vue",
   "main": "index.js",
   "files": [
-    "dist",
-    "index.js",
+    "charts-view.common.js",
+    "charts-view.umd.js",
+    "charts-view.umd.min.js",
```
Comment (theiliad) [inline packages/vue/package.json]: ...-vue?

Comment (cal-smith) [reply packages/vue/package.json]: ahahaha, yeah I actually meant `-vue`

Comment (cal-smith) [reply packages/vue/package.json]: You would not believe the number of times I've don that

Comment (theiliad) [reply packages/vue/package.json]: 😄

File: packages/angular/package.json [hunk 2]
```diff
@@ -19,39 +19,31 @@
...
+		"@angular/core": "6.1.0",
+		"@angular/platform-browser": "6.1.0",
+		"@angular/platform-browser-dynamic": "6.1.0",
+		"@babel/core": "7.3.4",
+		"@storybook/angular": "5.0.3",
 		"@types/node": "8.5.2",
-		"ng-packagr": "4.4.0",
-		"typescript": "2.7.2"
+		"babel-loader": "8.0.5",
+		"d3": "4.11.0",
```
Comment (theiliad) [inline packages/angular/package.json]: d3 & babel-loader are need for the angular package?

File: packages/core/src/polyfills.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
 (function() {
-	if (typeof window["CustomEvent"] === "function") return false;
+	if (typeof window["CustomEvent"] === "function") { return false; }
 
 	function CustomEvent(event, params) {
 		params = params || { bubbles: false, cancelable: false, detail: undefined };
-		var evt = document.createEvent("CustomEvent");
+		const evt = document.createEvent("CustomEvent");
```
Comment (theiliad) [inline packages/core/src/polyfills.ts]: 👍

File: packages/react/rollup.config.js [hunk 4]
```diff
@@ -1,17 +1,20 @@
 import babel from "rollup-plugin-babel";
-import commonjs from "rollup-plugin-commonjs";
 
 module.exports = {
     input: "./src/index.js",
     output: {
-        file: "./dist/bundle.js",
-        format: "cjs" 
+        file: "./dist/bundle/bundle.js",
```
Comment (theiliad) [inline packages/react/rollup.config.js]: is there an extra space here? the indenting doesn't look right on github

File: scripts/build-packages-and-demos.sh [hunk 5]
```diff
@@ -1,35 +1,36 @@
...
+# React build
+cd ../react
 npm run build
 npm run build-storybook
 cp -a storybook-dist/. ../../pages/react
 
-# Build Vue demos
+# Vue build
 cd ../vue
 npm run build
```
Comment (theiliad) [inline scripts/build-packages-and-demos.sh]: If we're going to have a `build` script in every package, then we can run `lerna run build` and it'll build all, and so on for the other standard commands as well

**Comment by cal-smith** [comment]
Fixed 👍

File: scripts/build-packages-and-demos.sh (hunk 5 above)
Comment (cal-smith) [reply scripts/build-packages-and-demos.sh]: That would definitely simplify things

Comment (theiliad) [reply scripts/build-packages-and-demos.sh]: 👍

Comment (cal-smith) [reply scripts/build-packages-and-demos.sh]: The demo build is `build-storybook` and `demo:build` ... if we standardize on one (`demo:build` I think) that'll work too

Comment (cal-smith) [reply scripts/build-packages-and-demos.sh]: Each demo should be able to hoist itself to the top level pages directory too 🤔

Comment (theiliad) [reply scripts/build-packages-and-demos.sh]: for sure, this whole file can be shrunk to the a few lines

Comment (cal-smith) [reply scripts/build-packages-and-demos.sh]: Or that can just be a separate collect step in the deploy script ... that's probably cleaner

File: packages/react/rollup.config.js (hunk 4 above)
Comment (cal-smith) [reply packages/react/rollup.config.js]: Probably ... I'm guessing the formatting was hit with some weird copypasta

File: packages/core/src/polyfills.ts (hunk 3 above)
Comment (cal-smith) [reply packages/core/src/polyfills.ts]: Thank tslint!

File: packages/angular/package.json (hunk 2 above)
Comment (cal-smith) [reply packages/angular/package.json]: babel-loader for storybook iirc ... let me check on d3 🤔

File: scripts/build-packages-and-demos.sh (hunk 5 above)
Comment (theiliad) [reply scripts/build-packages-and-demos.sh]: Pretty sure using `lerna run` you can have a script in root run in every package directory, that'll just regex all these folders and copy them over to `root/pages`

Comment (cal-smith) [reply scripts/build-packages-and-demos.sh]: Yep! `lerna exec` + some mild demo dir changes and it works super well 🎉

File: packages/angular/package.json (hunk 2 above)
Comment (cal-smith) [reply packages/angular/package.json]: It seems to be fine ... but it's better to explicitly declare our dependencies (d3 is a peer for core)

**Comment by cal-smith** [comment]
fixed!

File: scripts/build-packages-and-demos.sh [hunk 6]
```diff
@@ -1,39 +1,23 @@
...
-npm run build-storybook
-cp -a storybook-dist/. ../../pages/vue
 
-# Go back to project root folder
-cd ../..
+# run the build and demo:build script in all packages
+lerna run --parallel build
+lerna run --parallel demo:build
+# copy all the demos/{package name here} folders to the pages deploy directory
+lerna exec -- cp -a demo/bundle/. \$LERNA_ROOT_PATH/pages
```
Comment (theiliad) [inline scripts/build-packages-and-demos.sh]: 🥇

**Comment by cal-smith** [comment]
I'm pretty sure `netlify.sh` is up to date? I swapped it to run `npm run build-all`

**Comment by theiliad** [comment]
Yeah that's what I wanted, the `npm` command

**Pull request merged**